contribute.md

```diff
@@ -3,12 +3,251 @@ layout: page
 title: Contribute
 ---
 
-Each subsystem is separated into individual repositories within the linux-system-roles GitHub project. Just open a new issue against the appropriate subsystem’s issue tracker to report bugs or request enhancements at the subsytem listed in
+Each subsystem is separated into individual repositories within the
+[linux-system-roles](https://github.com/linux-system-roles) GitHub project.
+Just open a new issue against the appropriate subsystem's issue tracker to
+report bugs or request enhancements.  New subsystem requests or feedback can be
+provided to the project's landing page at
+[linux-system-roles.github.io](https://linux-system-roles.github.io) Pull
+requests welcome!
 
-[https://github.com/linux-system-roles](https://github.com/linux-system-roles)
 
-New subsystem requests or feedback can be provided to the project’s landing page at linux-system-roles.github.io Pull requests welcome!
+## Code structure
 
-The subsystems are licensed under different licences. By submitting a patch for inclusion, you are agreeing to license your changes under the subsystem licence.
+The repository is structured as follows:
 
-For discussions and questions, please use our mailing list (archives) or join our IRC channel #systemroles on freenode.
+- `./defaults/` - Contains the default role configuration.
+
+- `./examples/` - Contains YAML examples for different configurations.
+
+- `./meta/` - Metadata of the project.
+
+- `./tasks/` - Declaration of the different tasks that the role is going to execute.
+
+- `./tests/playbooks/` - Contains the complete tests for the role. `./tests/test_*.yml`
```

Comment:
Note that this is `tests/playbooks` is primarily used by the network role.  Every other role uses `tests/tests_*.yml` as the main playbook for the test and has no `playbooks` subdirectory.